operator/cmd/cilium_node.go

```diff
@@ -145,16 +145,29 @@ func startSynchronizingCiliumNodes(ctx context.Context, clientset k8sClient.Clie
 	// memory because 'ciliumNodeStore' is used across the operator
 	// to get the latest state of a CiliumNode.
 	if withKVStore || nodeManager != nil {
+		addToNodeManagerQueue := func(key string) {
+			if nodeManager != nil {
+				// Only enqueue key if already synced, handle at once while initializing
+				select {
+				case <-k8sCiliumNodesCacheSynced:
+					nodeManagerQueue.Add(key)
+				default:
+					err = nodeManagerSyncHandler(key)
+					if err != nil {
+						log.WithError(err).Errorf("initial sync %q failed with %v", key, err)
+					}
+				}
```

Comment:
@dylandreimerink I think in order to preserve the new behavior, which is more ideal with rate limiting / queuing, Sebastian's idea in https://github.com/cilium/cilium/pull/21526#issuecomment-1267044210 seems like it would be a better way to solve this. I can imagine that in large clusters where the Operator has to sync all CiliumNodes, potentially 100s, blocking the informer is something we should avoid when possible.